### PR TITLE
Add Typography Tokens from @metamask/design-system Package to twrnc settings

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
This PR contains the following changes
- `@metamask/storybook-react-native`
    - Added `EuclidCircularB` font and setup 
- `@metamask/design-system-react-native`
    - Updated sample story to render typography twrnc classes
- `@metamask/design-system-twrnc-preset`
    - Refactored color-related information to its own files
    - Added typography-related twrnc implementation

## **Related issues**

Fixes: #150 

## **Manual testing steps**

1. Go to root
2. Run `yarn install`
3. Run `yarn build`
4. Run `yarn storybook:ios`

### **After**
https://github.com/user-attachments/assets/34124aa6-6731-4406-86af-80eed6618ddf

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs)
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: packages/design-system-react-native/src/components/Button.stories.tsx [hunk 1]
```diff
@@ -18,8 +18,12 @@ export default ButtonMeta;
 export const Basic: StoryObj<typeof Button> = {
   render: () => {
     const tw = useTailwind();
+
     return (
       <View style={tw`bg-icon-default p-3`}>
+        <Text style={tw`text-display-md font-display-md text-primary-inverse`}>
+          Sample Text
+        </Text>
```
Comment (brianacnguyen) [inline packages/design-system-react-native/src/components/Button.stories.tsx:26]: Added simple addition to sample story to render changes before `Text` is merged

File: packages/design-system-twrnc-preset/src/Theme/Theme.constants.ts
Comment (brianacnguyen) [inline packages/design-system-twrnc-preset/src/Theme/Theme.constants.ts:1]: Moved to specific color/typography files

File: packages/design-system-twrnc-preset/src/Theme/Theme.utilities.ts [hunk 2]
```diff
@@ -1,24 +1,56 @@
...
+ * //       lineHeight: {
+ * //         'display-md': '40',
+ * //         // ...other line heights
+ * //       },
+ * //     },
+ * //   },
+ * //   plugins: [],
+ * // }
+ * @throws Will log an error and return an empty object if theme colors are not found for the specified color set and color scheme.
+ */
```
Comment (brianacnguyen) [inline packages/design-system-twrnc-preset/src/Theme/Theme.utilities.ts:53]: Added more comments to help understand utility function

File: packages/design-system-twrnc-preset/src/Theme/Theme.utilities.ts [hunk 3]
```diff
@@ -41,7 +71,19 @@ export const generateTailwindConfig = (
...
+          ...typographyTailwindConfig.fontSize,
+        },
+        fontFamily: {
+          ...typographyTailwindConfig.fontFamily,
+        },
+        letterSpacing: {
+          ...typographyTailwindConfig.letterSpacing,
+        },
+        lineHeight: {
+          ...typographyTailwindConfig.lineHeight,
```
Comment (brianacnguyen) [inline packages/design-system-twrnc-preset/src/Theme/Theme.utilities.ts:86]: @georgewrmarshall let me know if you prefer it this way or just `...typographyTailwindConfig`. This way prevents ANYTHING to be added to `typographyTailwindConfig` which may affect the final config if not done correctly

File: packages/design-system-twrnc-preset/src/twrnc-settings/typography.utilities.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ * getFontFamilyFromWeightAndStyle('400', 'normal');
+ * // Returns: 'EuclidCircularB-Regular'
+ *
+ * getFontFamilyFromWeightAndStyle('700', 'italic');
+ * // Returns: 'EuclidCircularB-BoldItalic'
+ *
+ * getFontFamilyFromWeightAndStyle('500');
+ * // Returns: 'EuclidCircularB-Medium'
+ */
+export const getFontFamilyFromWeightAndStyle = (
```
Comment (brianacnguyen) [inline packages/design-system-twrnc-preset/src/twrnc-settings/typography.utilities.ts:22]: Carried over from `metamask-mobile` https://github.com/MetaMask/metamask-mobile/blob/main/app/component-library/components/Texts/Text/Text.utils.ts

File: apps/storybook-react-native/assets/fonts/EuclidCircularB-Bold.otf
Comment (georgewrmarshall) [inline apps/storybook-react-native/assets/fonts/EuclidCircularB-Bold.otf:1]: Created an issue so we can consolidate the font files in future https://github.com/MetaMask/metamask-design-system/issues/153

File: packages/design-system-react-native/src/components/Button.stories.tsx (hunk 1 above)
Comment (georgewrmarshall) [reply packages/design-system-react-native/src/components/Button.stories.tsx:26]: Should this be rendering display-md style text?

Screenshot 2024-11-26 at 9 15 21 AM

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Checked storybook locally and all the typography variants 🚀 

Screenshot 2024-11-26 at 9 29 12 AM

File: packages/design-system-twrnc-preset/src/Theme/Theme.utilities.ts [hunk 5]
```diff
@@ -1,24 +1,56 @@
...
+import type { ColorSet } from '../twrnc-settings';
+import { colorSetList, typographyTailwindConfig } from '../twrnc-settings';
 
+/**
+ * Generates a Tailwind CSS configuration object based on the specified color set and color scheme.
+ * This configuration extends the base Tailwind settings with custom theme colors, typography settings,
+ * and other style properties for use in React Native with `twrnc`.
+ *
+ * @param colorSet - The color set to use (e.g., 'Brand', 'Neutral'). Determines the base palette for the theme.
+ * @param colorScheme - The current color scheme ('light' or 'dark'). Specifies whether to use light or dark mode styles.
```
[the remark this answers is not in the document]
Comment (brianacnguyen) [reply packages/design-system-twrnc-preset/src/Theme/Theme.utilities.ts:14]: Good spotting. `light` or `dark` follows how most systems name their `colorScheme`. I can update the `colorSet` to be consistent

File: packages/design-system-twrnc-preset/src/twrnc-settings/colors.types.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+/**
+ * Enum for different color set options.
+ */
+export enum ColorSet {
+  Brand = 'brand',
+}
+
```
Comment (brianacnguyen) [inline packages/design-system-twrnc-preset/src/twrnc-settings/colors.types.ts:7]: Add ColorScheme enum

**Review by georgewrmarshall — APPROVED** [review]
LGTM!

**Pull request merged**